src/Documentation/grains/actor_identity.md

```diff
@@ -0,0 +1,123 @@
+---
+layout: page
+title: Actor Identity
+---
+
+# Actor Identity
+
+In object-oriented environments, the identity of an object is hard to distinguish from a reference to it.
+Thus, when an object is created using new, the reference you get back represents all aspects of its identity except those that map the object to some external entity that it represents.
+
+In distributed systems, object references cannot represent instance identity, since references are typically limited to a single address space.
+That is certainly the case for .NET references.
+Furthermore, a virtual actor must have an identity regardless of whether it is active, so that we can activate it on demand.
+Therefore grains have a primary key.
+The primary key can be either a GUID (A Globally Unique Identifier), a long integer, or a string.
+
+The primary key is scoped to the grain type.
+Therefore, the complete identity of a grain is formed from the actor type and its key.
+
+The caller of the grain decides a long, a GUID, or a string scheme should be used.
+In fact the underlying data is the same, so the schemes can be used interchangeably.
+When a long is used, a GUID is actually created, and padded with zeros.
+
+Situations that require a singleton grain instance, such as a dictionary or registry, benefit from using `0` (a valid GUID) as its key.
```

Comment:
This line seems dated - maybe it should say `Guid.Empty` instead of `0`.

Reply:
The line now reads:

Situations that require a singleton grain instance, such as a dictionary or registry, benefit from using `Guid.Empty` as its key.

I took out the '(a valid GUID)' clause because I figured that it was explaining that '0' is a valid GUID - and now '0' is gone.